### ISSUE TITLE
[Feature Request] Common constants in the torch.* namespace

### ISSUE THREAD

**Issue by soumith** [pytorch/pytorch#19172]
## 🚀 Feature

Two feature requests asked for constants from the `math` package to be included in `torch` as well:
- https://github.com/pytorch/pytorch/issues/19123
- https://github.com/pytorch/pytorch/issues/6510

## Motivation

- To avoid adding an `import math` when you want to do it
- Because NumPy and SciPy do it

## Alternatives

Dont do it and tell users to `import math`

**Comment by soumith** [comment]
My opinion is that the convenience is not worth the replication, but I can see how going back to the top of the file and adding `import math` might seem inconvenient.

Additionally, with @umanwizard 's upcoming `torch.numpy`, it seems inevitable.

Any opinions?

**Comment by vadimkantorov** [comment]
If this is accepted, then I suggest also allowing functions like `torch.log` or `torch.ceil` to work on plain python scalars

**Comment by chausies** [comment]
@vadimkantorov not sure how that relates to this in particular.... But for that matter, I suggest literally all basic functions (like `torch.sin`) should all work on python scalars (like how numpy does). This issue already exists and is open, though, so I suggest pushing for it there: #11695

**Comment by vadimkantorov** [comment]
Agreed, let's track it there, my bad - didn't search existing issues this time. The reason I brought this up is that potential reason for existence for supporting `torch.log(10)` seemed similar (to me) to supporting `torch.pi`: the `math` alternative works fine, but supporting it saves a few keystrokes of the import.

**Comment by xuhdev** [comment]
I think it might worth the effort for the sake of the C++ interface (there is no reliable ways to determine the existence of a PI constant in C++, according to [this](https://stackoverflow.com/a/49778493/1150462), especially for float64 types). Plus, a git grep on PyTorch codebase reveals that multiple `PI`s are defined within the codebase, so at least a struct that stores common constants is desirable.
